### PR TITLE
cmd/gb-vendor/vendor: add support for repo.com/path.git format

### LINKED ISSUE #182: gb vendor fetch git.eclipse.org/gitroot/paho/org.eclipse.paho.mqtt.golang.git fail
```go get git.eclipse.org/gitroot/paho/org.eclipse.paho.mqtt.golang.git```
works, but

```gb vendor fetch git.eclipse.org/gitroot/paho/org.eclipse.paho.mqtt.golang.git
FATAL command "git.eclipse.org/gitroot/paho/org.eclipse.paho.mqtt.golang.git" failed: unknown repository type
FATAL command "vendor" failed: exit status 1```

### PR DESCRIPTION
**Pull request by davecheney**
Fixes #182

### CODE REVIEW

**Comment by davidkaya** [comment]
Is the source code formatted or github messed it up for me?

**Comment by davecheney** [comment]
looks ok to me, what are you seeing ?

**Comment by davidkaya** [comment]
Line 58, 89, those curly brackets seem off to me.

**Comment by davecheney** [comment]
They look fine to me, try pulling the branch and running go fmt

**Comment by davidkaya** [comment]
Well, that is weird. In Intellij it looks good, in browser: 
screen shot 2015-06-09 at 13 52 19

File: cmd/gb-vendor/vendor/repo.go [hunk 1]
```diff
@@ -85,27 +86,46 @@ func DeduceRemoteRepo(path string) (RemoteRepo, string, error) {
...
+			return repo, v[6], err
 		default:
-			return nil, "", fmt.Errorf("unknown repository type: %q", vcs)
+			return nil, "", fmt.Errorf("unknown repository type: %q", v[5])
+
 		}
 	}
+
+	// no idea, try to resolve as a vanity import
+	importpath, vcs, reporoot, err := ParseMetadata(path)
```
Comment (davidkaya) [inline cmd/gb-vendor/vendor/repo.go]: Daaaaaaaamn that's a lot of return values :smile:

**Pull request merged**